code/modules/mob/living/silicon/silicon.dm

```diff
@@ -211,19 +223,42 @@
 
 
 /mob/living/silicon/bullet_act(obj/item/projectile/Proj)
-
-
 	if(!Proj.nodamage)
+		var/damage = run_armor(Proj.damage, Proj.damage_type, Proj.flag, 0, Proj.armour_penetration_flat, Proj.armour_penetration_percentage)
 		switch(Proj.damage_type)
 			if(BRUTE)
-				adjustBruteLoss(Proj.damage)
+				adjustBruteLoss(damage)
 			if(BURN)
-				adjustFireLoss(Proj.damage)
+				adjustFireLoss(damage)
 
 	Proj.on_hit(src,2)
 
 	return 2
 
+/mob/living/silicon/attacked_by(obj/item/I, mob/living/user, def_zone)
+	send_item_attack_message(I, user)
+	if(I.force)
+		var/bonus_damage = 0
+		if(ishuman(user))
+			var/mob/living/carbon/human/H = user
+			bonus_damage = H.physiology.melee_bonus
+		var/damage = run_armor(I.force + bonus_damage, I.damtype, MELEE, 0, I.armour_penetration_flat, I.armour_penetration_percentage)
+		apply_damage(damage, I.damtype, def_zone)
+
+///returns the damage value of the attack after processing the silicons's various armor protections
+/mob/living/silicon/proc/run_armor(damage_amount, damage_type, damage_flag = 0, attack_dir, armour_penetration_flat = 0, armour_penetration_percentage = 0)
+	switch(damage_type)
+		if(BRUTE)
+		if(BURN)
+		else
+			return 0
```

Comment:
was there supposed to be something here?

Reply:
It's a switch, so no I think. It's checking if it's brute or burn, which will lead to damage, any other source is no damage so no need to check armor.

Reply:
i'd just do `if(damage_type != BRUTE && damage_type != BURN)`